File: anagram-of-palindrome/anagramofpalindrome.py
"""Is the word an anagram of a palindrome?

A palindrome is a word that reads the same forward and backwards
(eg, "racecar", "tacocat"). An anagram is a rescrambling of a word
(eg for "racecar", you could rescramble this as "arceace").

Determine if the given word is a re-scrambling of a palindrome.
The word will only contain lowercase letters, a-z.

Examples::

    >>> is_anagram_of_palindrome("a")
    True

    >>> is_anagram_of_palindrome("ab")
    False

    >>> is_anagram_of_palindrome("aab")
    True

    >>> is_anagram_of_palindrome("arceace")
    True

    >>> is_anagram_of_palindrome("arceaceb")
    False

"""


def is_anagram_of_palindrome(word):
    """Is the word an anagram of a palindrome?"""

    letters_dict = {}

    for letter in word:
        if letter in letters_dict:
            letters_dict[letter] += 1
        else:
            letters_dict[letter] = 1

    dict_contains_odd_value = False

    for key in letters_dict:
        if letters_dict[key] % 2 == 0:
            continue
        elif dict_contains_odd_value == False:
            dict_contains_odd_value = True
        else:
            return False

    return True



if __name__ == '__main__':
    import doctest

    if doctest.testmod().failed == 0:
        print("\n*** ALL TEST PASSED. AWESOMESAUCE!\n")

File: anagram-of-palindrome/anagramofpalindrome.js
/*Is the word an anagram of a palindrome?

A palindrome is a word that reads the same forward and backwards
(eg, "racecar", "tacocat"). An anagram is a rescrambling of a word
(eg for "racecar", you could rescramble this as "arceace").

Determine if the given word is a re-scrambling of a palindrome.
The word will only contain lowercase letters, a-z.

Examples::

    >>> is_anagram_of_palindrome("a")
    True

    >>> is_anagram_of_palindrome("ab")
    False

    >>> is_anagram_of_palindrome("aab")
    True

    >>> is_anagram_of_palindrome("arceace")
    True

    >>> is_anagram_of_palindrome("arceaceb")
    False
    */

function isAnagramOfPalindrome(string) {
    
    letter_counts = {}

    for ( i = 0; i < string.length; i++){
        letter = string[i]
        if (typeof letter_counts[letter] === 'number'){
            letter_counts[letter] += 1;
        }
        else {
            letter_counts[letter] = 1;
        }
    }

    let letter_with_odd_count = false;

    for (let key in letter_counts){
        if (letter_counts[key] % 2 != 0){
            if (letter_with_odd_count === false){
                letter_with_odd_count = true;
            } else {
                return false;
            }
        }

    }
    return true;

}